easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1226,6 +1226,31 @@ def get_parsed_multi_deps(self):
 
         return multi_deps
 
+    def find_dep_version_match(self, dep):
+        """Identify the correct version for this system from the choices provided. This returns the version to use."""
+        if isinstance(dep['version'], string_type):
+            self.log.debug("Version is already a string ('%s'), OK", dep['version'])
+            return dep['version']
+        elif dep['version'] is None:
+            self.log.debug("Version is None, OK", dep)
+            return None
+        elif isinstance(dep['version'], dict):
+            # figure out matches based on dict keys (after splitting on '=')
+            my_arch_key = 'arch=%s' % get_cpu_architecture()
+            arch_keys = [x for x in dep['version'].keys() if x.startswith('arch=')]
+            other_keys = [x for x in dep['version'].keys() if '=' in x and x not in arch_keys]
```

Comment:
@edmondac I think we should drop the `'=' in x` part.

If I use the following:

```python
dependencies = [('example', {'foo': '1.2.3'}]
```

Then you should still get an error for `Unexpected keys`, even though `foo` doesn't include `=`?

Reply:
done